<?xml version="1.0" encoding="utf-8"?>
<!-- This URDF was automatically created by SolidWorks to URDF Exporter! Originally created by Stephen Brawner (brawner@gmail.com) 
     Commit Version: 1.6.0-1-g15f4949  Build Version: 1.6.7594.29634
     For more information, please see http://wiki.ros.org/sw_urdf_exporter -->
<robot
  name="rpr_arm">

  <link name="world"/>
  

  <joint name="base_joint" type="fixed">
    <parent link="world"/>
    <child link="base_link"/>
    <origin xyz="0.0 0.0 0.199" rpy="0.0 0.0 0.0"/>
  </joint>
  
  <link
    name="base_link">
    <inertial>
      <origin
        xyz="2.9724E-06 -0.095252 -2.2887E-06"
        rpy="0 0 0" />
      <mass
        value="36.091" />
      <inertia
        ixx="0.20075"
        ixy="5.0161E-07"
        ixz="2.116E-09"
        iyy="0.13103"
        iyz="2.782E-06"
        izz="0.20075" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://rpr_arm/meshes/base_link.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="1 0 0 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://rpr_arm/meshes/base_link.STL" />
      </geometry>
    </collision>
  </link>
  <link
    name="link_1">
    <inertial>
      <origin
        xyz="-1.2862E-05 0.037934 -0.0025981"
        rpy="0 0 0" />
      <mass
        value="1.7235" />
      <inertia
        ixx="0.0016319"
        ixy="-8.0768E-07"
        ixz="5.8946E-08"
        iyy="0.001175"
        iyz="0.00012725"
        izz="0.0016044" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://rpr_arm/meshes/link_1.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0 1 0 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://rpr_arm/meshes/link_1.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="joint_1"
    type="revolute">
    <origin
      xyz="0 0 0.031"
      rpy="1.5708 0 0" />
    <parent
      link="base_link" />
    <child
      link="link_1" />
    <axis
      xyz="0 1 0" />
    <limit
      lower="-2.61"
      upper="2.61"
      effort="20"
      velocity="2" />
  </joint>
  <link
    name="link_2">
    <inertial>
      <origin
        xyz="-3.605E-18 0.1356 0.025302"
        rpy="0 0 0" />
      <mass
        value="3.8974" />
      <inertia
        ixx="0.028517"
        ixy="-2.0064E-19"
        ixz="2.3982E-20"
        iyy="0.0013623"
        iyz="0.00050471"
        izz="0.02872" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://rpr_arm/meshes/link_2.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0 1 0 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://rpr_arm/meshes/link_2.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="joint_2"
    type="revolute">
    <origin
      xyz="0 0.075 0.006"
      rpy="0 0 -0.00066622" />
    <parent
      link="link_1" />
    <child
      link="link_2" />
    <axis
      xyz="0 0 1" />
    <limit
      lower="-1.5701"
      upper="1.5715"
      effort="20"
      velocity="2" />
  </joint>
  <link
    name="link_3">
    <inertial>
      <origin
        xyz="-0.085488 2.0166E-16 -0.027458"
        rpy="0 0 0" />
      <mass
        value="2.6549" />
      <inertia
        ixx="0.0078884"
        ixy="8.2963E-18"
        ixz="-9.6143E-06"
        iyy="0.011372"
        iyz="-1.4951E-19"
        izz="0.0046215" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://rpr_arm/meshes/link_3.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0 0 1 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://rpr_arm/meshes/link_3.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="joint_3"
    type="revolute">
    <origin
      xyz="0 0.29 0.03"
      rpy="3.1416 0 -1.5708" />
    <parent
      link="link_2" />
    <child
      link="link_3" />
    <axis
      xyz="0 0 1" />
    <limit
      lower="-2.0944"
      upper="2.1817"
      effort="20"
      velocity="2" />
  </joint>
  <link
    name="gripper_left">
    <inertial>
      <origin
        xyz="-8.0344E-06 0.024881 -0.011816"
        rpy="0 0 0" />
      <mass
        value="0.31443" />
      <inertia
        ixx="0.00013189"
        ixy="-1.289E-08"
        ixz="1.5157E-09"
        iyy="5.6101E-05"
        iyz="1.168E-05"
        izz="0.0001712" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://rpr_arm/meshes/gripper_left.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0 1 1 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://rpr_arm/meshes/gripper_left.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="grip_left"
    type="prismatic">
    <origin
      xyz="-0.126 0 0.0"
      rpy="0 0 1.5708" />
    <parent
      link="link_3" />
    <child
      link="gripper_left" />
    <axis
      xyz="0 0 1" />
    <limit
      lower="0"
      upper="0.075"
      effort="50"
      velocity="0.5" />
    <mimic
      joint="grip_right"
      multiplier="1"
      offset="0" />
  </joint>
  <link
    name="gripper_right">
    <inertial>
      <origin
        xyz="-8.0344E-06 0.024881 0.008184"
        rpy="0 0 0" />
      <mass
        value="0.31443" />
      <inertia
        ixx="0.00013189"
        ixy="-1.289E-08"
        ixz="1.5159E-09"
        iyy="5.6101E-05"
        iyz="1.168E-05"
        izz="0.0001712" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://rpr_arm/meshes/gripper_right.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0 1 1 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://rpr_arm/meshes/gripper_right.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="grip_right"
    type="prismatic">
    <origin
      xyz="-0.126 0 0.0"
      rpy="3.1416 0 -1.5708" />
    <parent
      link="link_3" />
    <child
      link="gripper_right" />
    <axis
      xyz="0 0 1" />
    <limit
      lower="0"
      upper="0.07"
      effort="50"
      velocity="0.5" />
  </joint>

  <transmission name="trans_joint_1">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint_1">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </joint>
        <actuator name="joint_1_motor">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
  </transmission>
  
  <transmission name="trans_joint_2">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint_2">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </joint>
        <actuator name="joint_2_motor">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
  </transmission>
  
  <transmission name="trans_joint_3">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint_3">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </joint>
        <actuator name="joint_3_motor">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
  </transmission>
  
  <transmission name="trans_gripper_joint_1">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="grip_right">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </joint>
        <actuator name="grip_right_motor">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
  </transmission>
  
  <transmission name="trans_gripper_joint_2">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="grip_left">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </joint>
        <actuator name="grip_left_motor">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
  </transmission>
  
  <gazebo>
      <plugin name="gazebo_ros_control" filename="libgazebo_ros_control.so">
        <robotNamespace>/rpr_arm</robotNamespace>
      </plugin>

      <plugin name="gripper_joint_mimic" filename="librobotics_gazebo_mimic_joint_plugin.so">
        <joint name="grip_right" />
        <mimicJoint name="grip_left" />
        <multiplier value="1" />
        <offset value="0" />
      </plugin>
  </gazebo>
</robot>